fix: Show sale bage on Product Image Gallery block

gigitux opened the pull request:
The new Product Image Gallery block was not displaying the badge for a product that is on sale. This PR fixes this bug.

<!-- Reference any related issues or PRs here -->

Fixes #8528

<!-- Don't forget to update the title with something descriptive. -->
<!-- If you can, add the appropriate labels -->


### Testing

#### User Facing Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. Go to Appearance > Themes and activate a Blockfied theme, such as: Twenty-twenty Three;
2. After the theme is activated, go to Appearance > Editor (Beta);
3. Inside the Design section, click on Templates;
4. Select the Single Product template;
5. On the top left of the page, click on the Edit button;
6. Add the Product Image Gallery block to the editor;
7. Go to your website, and open a Product with a On Sale status.
8. Look at the top left of the Product Image Gallery and check that the badge appears.


* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

gigitux committed:
fix: show sale bage on Product Image Gallery block #8528

thealexandrelara reviewed (commented):
Thank you for working and solving the issue. There is one thing that I'd like to check with you before approving the PR. 

Currently, we can have multiple Product Image Gallery blocks added to the page, so I added more than one to the page inside a Group block and I realized the Sale badges are stacking at the top (probably caused by the fact that the `span.onsale` class has a `position: absolute`), like this:

![image](https://user-images.githubusercontent.com/20469356/223539787-f9d44660-8e14-49ca-8d96-22b650d84e5b.png)


Do you think this is an issue we have to solve or it would be better to just limit to 1 the amount of product image gallery blocks that we can add to the page?

gigitux commented:
> Thank you for working and solving the issue. There is one thing that I'd like to check with you before approving the PR.
> 
> Currently, we can have multiple Product Image Gallery blocks added to the page, so I added more than one to the page inside a Group block and I realized the Sale badges are stacking at the top (probably caused by the fact that the `span.onsale` class has a `position: absolute`), like this:
> 
> ![image](https://user-images.githubusercontent.com/20469356/223539787-f9d44660-8e14-49ca-8d96-22b650d84e5b.png)
> 
> Do you think this is an issue we have to solve or it would be better to just limit to 1 the amount of product image gallery blocks that we can add to the page?

Great catch! For now, let's limit it to one! I think that we should do this for all blocks related to the Single Product template.

Fixed via 97b6a7245f7ed16bd938c916ad648b1582f5a50f

gigitux committed:
Merge branch 'trunk' of https://github.com/woocommerce/woocommerce-blocks into fix/8528-product-image-gallery-block-not-showing-sale-badge

gigitux committed:
set multiple flag to false

thealexandrelara reviewed (approved):
Just tested and confirmed the issue is no longer happening since now the block limit was set to just 1. Thank you for solving it! ✅

gigitux committed:
Merge branch 'trunk' into fix/8528-product-image-gallery-block-not-showing-sale-badge
